examples/lstm_text_generation.py

```diff
@@ -0,0 +1,230 @@
+    updates = lasagne.updates.adagrad(cost, all_params, LEARNING_RATE)
+
+    # Theano functions for training and computing cost
+    print("Compiling functions ...")
+    train = theano.function([l_in.input_var, target_values], cost, updates=updates, allow_input_downcast=True)
+    compute_cost = theano.function([l_in.input_var, target_values], cost, allow_input_downcast=True)
+
+    # In order to generate text from the network, we need the probability distribution of the next character given
+    # the state of the network and the input (a seed).
+    # In order to produce the probability distribution of the prediction, we compile a function called probs. 
+    
+    probs = theano.function([l_in.input_var],network_output,allow_input_downcast=True)
+
+    # The next function generates text given a phrase of length at least SEQ_LENGTH.
+    # The phrase is set using the variable generation_phrase.
+    # The optional input "N" is used to set the number of characters of text to predict. 
+
+    def try_it_out(N=200):
+        '''
+        This function uses the user-provided string "generation_phrase" and current state of the RNN generate text.
+        The function works in three steps:
+        1. It converts the string set in "generation_phrase" (which must be over SEQ_LENGTH characters long) 
+           to encoded format. We use the gen_data function for this. By providing the string and asking for a single batch,
+           we are converting the first SEQ_LENGTH characters into encoded form. 
+        2. We then use the LSTM to predict the next character and store it in a (dynamic) list sample_ix. This is done by using the 'probs'
+           function which was compiled above. Simply put, given the output, we compute the probabilities of the target and pick the one 
+           with the highest predicted probability. 
+        3. Once this character has been predicted, we construct a new sequence using all but first characters of the 
+           provided string and the predicted character. This sequence is then used to generate yet another character.
+           This process continues for "N" characters. 
+        To make this clear, let us again look at a concrete example. 
+        Assume that SEQ_LENGTH = 5 and generation_phrase = "The quick brown fox jumps". 
+        We initially encode the first 5 characters ('T','h','e',' ','q'). The next character is then predicted (as explained in step 2). 
+        Assume that this character was 'J'. We then construct a new sequence using the last 4 (=SEQ_LENGTH-1) characters of the previous
+        sequence ('h','e',' ','q') , and the predicted letter 'J'. This new sequence is then used to compute the next character and 
+        the process continues.
+        '''
+
+        assert(len(generation_phrase)>=SEQ_LENGTH)
+        sample_ix = []
+        x,_ = gen_data(0, 1, generation_phrase)
+
+        for i in range(N):
+            ix = np.argmax(probs(x).ravel())
```

Comment:
@skaae, you're right. That is another way of doing it. I was trying to mimic the [Keras](https://github.com/fchollet/keras/blob/master/examples/lstm_text_generation.py) example. 

@f0k, that's how I had it when I first wrote the code. I was using `np.random.choice` instead of `np.argmax`. However, there seemed to be some NumPy issues that were cropping in (especially if the NumPy version was slightly older). I could write my own random choice but ultimately, I felt that `argmax` was interpretable and needed no code.